apps/src/code-studio/components/ExportDialog/IconPage.jsx

```diff
@@ -54,11 +55,26 @@ export default class IconPage extends React.Component {
             Upload another image
           </button>
         </div>
+        <input
+          ref={input => (this.uploadFileInput = input)}
+          type="file"
+          style={styles.hiddenFileInput}
+          accept="image/png"
+          onChange={this.handleSelectUploadFile}
+        />
       </div>
     );
   }
 
-  uploadIcon = () => {
-    // TODO: implement icon upload
+  handleSelectUploadFile = () => {
```

Comment:
Maybe I just need another ☕️, but it seems like there are too many ways to interpret this function name. Mind renaming or commenting? (i.e. It can be interpreted as `selectFile`, `uploadFile`, `selectFileToUpload` or `selectedUploadFileFromList`)

Reply:
This was adapted from another piece of code in our codebase. I agree that the name is confusing

I think I'll change it to the more obvious `onUploadFileInputChange` (since we're handling `onChange` for what we are calling the `uploadFileInput`)